src/pymovements/events/events.py

```diff
@@ -180,7 +180,15 @@ def __init__(
 
         # Ensure column order: trial columns, name, onset, offset.
         if self.trial_columns is not None:
-            self.frame = self.frame.select([*self.trial_columns, *self._minimal_schema.keys()])
+            minimal_cols = set(self._minimal_schema.keys())
+            trial_cols_set = set(self.trial_columns)
+            other_cols = [
+                col for col in self.frame.columns
+                if col not in trial_cols_set and col not in minimal_cols
```

Comment:
is there a reason why you don't just write this and leave out the variables above?

```suggestion
                if col not in [*self.trial_columns, *self._minimal_schema.keys()]
```